internal/metamorph/processor.go

```diff
@@ -382,6 +387,22 @@ func (p *Processor) StartSendStatusUpdate() {
 				return
 
 			case msg := <-p.statusMessageCh:
+				// if we receive new update check if we have client connection waiting for status and send it
+				found := p.responseProcessor.UpdateStatus(msg.Hash, StatusAndError{
+					Hash:         msg.Hash,
+					Status:       msg.Status,
+					Err:          msg.Err,
+					CompetingTxs: msg.CompetingTxs,
+				})
+
+				if !found {
+					found = p.txFoundInCache(msg.Hash)
+				}
+
+				if !found {
+					continue
+				}
+
```

Comment:
Please add here a debug log for the status message e.g.

```go
p.logger.Debug("Status update received", slog.String("hash", msg.Hash), slog.String("status", msg.Status))
```

Reply:
Done